File: proyecto-final/src/main/java/com/utn/tsp/proyectofinal/Model/FacturaDetalle.java
package com.utn.tsp.proyectofinal.Model;

import lombok.*;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Table(name = "FACTURA_DETALLE")
@Getter
@Setter
@Data
@AllArgsConstructor
@NoArgsConstructor
public class FacturaDetalle implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long facturaDetalleId;
    @ManyToOne()
    @JoinColumn(name = "facturaId")
    private Factura factura;
    @ManyToOne()
    @JoinColumn(name = "productoId")
    private Producto producto;
    private Double precioUnitario;
    private int cantidad;

}

File: gestor-ventas/src/app/models/facturaDetalle.ts
import { Factura } from "./factura";
import { Producto } from "./producto";

export interface FacturaDetalle{

    facturaDetalleId: number | any;
    factura: Factura | any;
    producto: Producto;
    precioUnitario: number;
    cantidad: number;

}